### ISSUE TITLE
Reduce failure rate of unit test testParquetFileWithBloomFilterWithFpp

### ISSUE THREAD

**Issue by asfimport** [apache/parquet-java#2824]
```java

[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    TestParquetWriter.testParquetFileWithBloomFilterWithFpp:342
[INFO]  
```
The unit test utilizes random string generation for test data without using a fixed seed. The expectation of a unit test is that the number of false positives in the Bloom filter should match the set probability. Therefore, a simple fix is to increase the number of tests on the Bloom filter. The reason for not using a fixed seed with random numbers is to avoid making the tests effective only in specific scenarios. If it is necessary to use a fixed seed, I can also modify the PR accordingly.

**Reporter**: [Feng Jiajie](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fengjiajie) / @fengjiajie
**Assignee**: [Feng Jiajie](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fengjiajie) / @fengjiajie

<sub>**Note**: *This issue was originally created as [PARQUET-2361](https://issues.apache.org/jira/browse/PARQUET-2361). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

**Comment by asfimport** [comment]
[Atour Mousavi Gourabi](https://issues.apache.org/jira/browse/PARQUET-2361?#comment-17774984) / @amousavigourabi:
@fengjiajie +1 for this issue, these flakes are quite annoying. I could also take a look if necessary.

**Comment by asfimport** [comment]
[Feng Jiajie](https://issues.apache.org/jira/browse/PARQUET-2361?#comment-17775115) / @fengjiajie:
Hi @amousavigourabi 

I found two issues that may cause the failures to occur easily:
1. This may be a bug. The '{**}distinctStrings{**}' is used to generate test files, but it is cleared in the first round. Then it is used for probability testing, so in the next round of generating 'fpp' files, it uses the data from the previous round of testing, and the length of the filled strings is also different (length 10 for test not in bloomfilter) is also different from the initial one (strings length 12 for build bloomfilter).
1. The number of test iterations is insufficient, resulting in an unstable probability.
   
   I attempted to submit a pull request：<https://github.com/apache/parquet-mr/pull/1170>
